### PR TITLE
Bug 1548743 - Improve styling of trailhead flow (scrollable page, animations, dismiss)

### PR DESCRIPTION
**Pull request by rlr**
* animation to reveal the cards after closing modal
* (subtle) animation to reveal modal content
* fix devtools toggle button position
* responsive tweaks to the modal content
* dismiss cards feature

### CODE REVIEW

File: content-src/asrouter/templates/Trailhead/_Trailhead.scss [hunk 1]
```diff
@@ -214,11 +232,32 @@
       outline: dotted 1px;
     }
   }
+
+  .trailheadInner,
+  .trailheadStart {
+    opacity: 0;
+    transition: opacity 0.4s;
```
Comment (Mardak) [inline content-src/asrouter/templates/Trailhead/_Trailhead.scss]: We don't need all the `revealModalContent` related state/methods/setTimeout as we only fade in (and no fade out) and already have a css animation handy:
https://github.com/mozilla/activity-stream/blob/5a8c8f68060ccdf7773502578eaa05559f438264/content-src/styles/_activity-stream.scss#L60-L66

Should be able to just do:
```css
.trailheadInner,
.trailheadStart {
  animation: fadeIn 0.4s;
}
```

File: content-src/asrouter/templates/Trailhead/_Trailhead.scss [hunk 2]
```diff
@@ -80,13 +86,21 @@
     padding: 0;
 
     li {
-      background-position: left 4px;
+      background-position: left 6px;
```
Comment (Mardak) [inline content-src/asrouter/templates/Trailhead/_Trailhead.scss]: nit: let's just set these individually `background-position-x` `background-position-y` and only change the latter for the breakpoint.

File: content-src/asrouter/templates/Trailhead/Trailhead.jsx [hunk 3]
```diff
@@ -185,6 +221,12 @@ export class _Trailhead extends React.PureComponent {
             {...card} />
         ))}
         </div>
+        {this.state.showCardPanel &&
+          <button
+            className="icon icon-dismiss" onClick={this.hideCardPanel}
+            title={props.intl.formatMessage({id: "menu_action_dismiss"})}
+            aria-label={props.intl.formatMessage({id: "menu_action_dismiss"})} />
```
Comment (Mardak) [inline content-src/asrouter/templates/Trailhead/Trailhead.jsx]: Good thing we have so many ways to get strings… 😉

File: content-src/asrouter/templates/Trailhead/_Trailhead.scss (hunk 2 above)
Comment (rlr) [reply content-src/asrouter/templates/Trailhead/_Trailhead.scss]: oh and handle RTL! 😉

Comment (rlr) [reply content-src/asrouter/templates/Trailhead/_Trailhead.scss]: nvm, you already did 😄

**Comment by rlr** [comment]
I am guessing that `unreferenced file: resource://gre/modules/LoginAutoCompleteResult.jsm` error isn't this patch

**Comment by rlr** [comment]
I snuck in some active state styles for the buttons:

```
diff --git a/content-src/asrouter/templates/Trailhead/_Trailhead.scss b/content-src/asrouter/templates/Trailhead/_Trailhead.scss
index 3fa73769..b2a3e5a4 100644
--- a/content-src/asrouter/templates/Trailhead/_Trailhead.scss
+++ b/content-src/asrouter/templates/Trailhead/_Trailhead.scss
@@ -209,6 +209,10 @@
       &:focus {
         outline: dotted 1px;
       }
+
+      &:active {
+        background-color: $trailhead-blue-70;
+      }
     }
   }

@@ -231,6 +235,10 @@
     &:focus {
       outline: dotted 1px;
     }
+
+    &:active {
+      background-color: $trailhead-blue-70;
+    }
   }

   .trailheadInner,
@@ -357,6 +365,14 @@
       box-shadow: none;
       background: var(--trailhead-card-button-background-hover-color);
     }
+
+    &:focus {
+      outline: dotted 1px;
+    }
+
+    &:active {
+      background: var(--trailhead-card-button-background-active-color);
+    }
   }

   .onboardingButtonContainer {
diff --git a/content-src/styles/_theme.scss b/content-src/styles/_theme.scss
index 595348dc..68ac4dac 100644
--- a/content-src/styles/_theme.scss
+++ b/content-src/styles/_theme.scss
@@ -83,6 +83,7 @@ body {
   --trailhead-cards-background-color: #{$grey-20};
   --trailhead-card-button-background-color: #{$grey-90-10};
   --trailhead-card-button-background-hover-color: #{$grey-90-20};
+  --trailhead-card-button-background-active-color: #{$grey-90-30};

   &[lwt-newtab-brighttext] {
     // General styles
@@ -148,5 +149,6 @@ body {
     --trailhead-cards-background-color: #{$grey-90-10};
     --trailhead-card-button-background-color: #{$grey-90-30};
     --trailhead-card-button-background-hover-color: #{$grey-90-40};
+    --trailhead-card-button-background-active-color: #{$grey-90-50};
   }
 }
diff --git a/content-src/styles/_variables.scss b/content-src/styles/_variables.scss
index 1469cc25..1cbfae7e 100644
--- a/content-src/styles/_variables.scss
+++ b/content-src/styles/_variables.scss
@@ -78,6 +78,7 @@ $firefox-wordmark-default-color: #363959;
 $firefox-wordmark-darktheme-color: $white;
 $trailhead-purple: #2B2156;
 $trailhead-blue-60: #0250BB;
+$trailhead-blue-70: #054096;

 // Photon transitions from http://design.firefox.com/photon/motion/duration-and-easing.html
 $photon-easing: cubic-bezier(0.07, 0.95, 0, 1);
```

**Pull request merged**